src/routers/axelar/XCMRouter.sol

```diff
@@ -0,0 +1,205 @@
+        _;
+    }
+
+    modifier onlyAxelarEVMRouterOrigin() {
+        require(
+            msg.sender == address(axelarEVMRouterOrigin),
+            "ConnectorAxelarXCMRouter/only-axelar-evm-router-origin-allowed"
+        );
+        _;
+    }
+
+    // --- Administration ---
+    function rely(address user) external auth {
+        wards[user] = 1;
+        emit Rely(user);
+    }
+
+    function deny(address user) external auth {
+        wards[user] = 0;
+        emit Deny(user);
+    }
+
+    function file(bytes32 what, address axelarEVMRouterOrigin_) external auth {
+        if (what == "axelarEVMRouterOrigin") {
+            axelarEVMRouterOrigin = axelarEVMRouterOrigin_;
+            emit File(what, axelarEVMRouterOrigin_);
+        } else {
+            revert("ConnectorAxelarXCMRouter/file-unrecognized-param");
+        }
+    }
+
+    function file(bytes32 what, uint64 buyExecutionWeightLimit, uint64 transactWeightAtMost, uint256 feeAmount)
+        external
+        auth
+    {
+        if (what == "xcmWeightInfo") {
+            xcmWeightInfo = XcmWeightInfo(buyExecutionWeightLimit, transactWeightAtMost, feeAmount);
+        } else {
+            revert("ConnectorAxelarXCMRouter/file-unrecognized-param");
+        }
+
+        emit File(what, xcmWeightInfo);
+    }
+
+    // --- Incoming ---
+    // A message that's coming from another EVM chain, headed to the Centrifuge Chain.
+    function execute(bytes32, string calldata sourceChain, string calldata, bytes calldata payload)
+        external
+        onlyAxelarEVMRouterOrigin
+    {
+        // todo(nuno): why do we hash this?
+        bytes32 hh = keccak256(payload);
+
+        XcmTransactorV2 transactorContract = XcmTransactorV2(XCM_TRANSACTOR_V2_ADDRESS);
+
+        transactorContract.transactThroughSignedMultilocation(
+            // dest chain
+            centrifuge_parachain_multilocation(),
+            // fee asset
+            cfg_asset_multilocation(),
+            // the weight limit for the transact call execution
+            xcmWeightInfo.transactWeightAtMost,
+            // the call to be executed on the cent chain
+            payload,
+            // the CFG we offer to pay for execution fees of the whole XCM
+            xcmWeightInfo.feeAmount,
+            // overall XCM weight, the total weight the XCM-transactor extrinsic can use.
+            // This includes all the XCM instructions plus the weight of the Transact call itself.
+            xcmWeightInfo.buyExecutionWeightLimit
+        );
+
+        executedCalls[hh] = 1;
+        emit Executed(hh);
+
+        return;
+    }
+
+    // --- Outgoing ---
+    // A message that has been sent from the Centrifuge Chain, heading to a specific destination EVM chain
+    function send(string calldata destinationChain, string calldata destinationAddress, bytes calldata payload)
+        external
+        payable
```

Comment:
The function is payable. What happens with the received funds?

Reply:
I think this bit as copy pasted; do we need to make it `payable`? and how can we control where the funds go to?

Reply:
we have multiple options to control the funds in the contract. 
I can not tell if we need payable here or not without understanding how the cross chain tx fee accounting works

Reply:
Working on understanding that. In Miguel's POC contract that inspired this one he used `payable`, so I guess that was necessary. Will keep you posted!

Reply:
I thought that maybe we needed the `payable` keyword due to implementing Axelar's `AxelarGatewayLike.callContract` interface function but that function is NOT `payable` and looking through Axelar's solidity codebase the only occasions where `payable` is used is around fee deposits and management where it makes sense to use it. Dropping it here.